frontend/appflowy_tauri/src/appflowy_app/components/board/Board.hooks.ts

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useState } from 'react';
+import { useAppDispatch, useAppSelector } from '../../stores/store';
+import { boardActions } from '../../stores/reducers/board/slice';
+import { ICellData, IDatabase, IDatabaseRow, ISelectOption } from '../../stores/reducers/database/slice';
+
+export const useBoard = () => {
+  const dispatch = useAppDispatch();
+  const groupingFieldId = useAppSelector((state) => state.board);
+  const database = useAppSelector((state) => state.database);
+  const [title, setTitle] = useState('');
+  const [boardColumns, setBoardColumns] =
+    useState<(ISelectOption & { rows: (IDatabaseRow & { isGhost: boolean })[] })[]>();
+  const [movingRowId, setMovingRowId] = useState<string | undefined>(undefined);
```

Comment:
I think using the Option<String> instead of `string | undefined` would be better.  The option concept comes from Rust and it was introduced in the ts-result package.

Reply:
in this particular case it's normal to use `string | undefined` because this variable(state) is used in UI. In other cases I will use Option and Result types